File: typescript-connector/src/models/Step.ts
// Class from DFC Business Ontology: #Step
import { SemanticObject } from "../core/SemanticObject.js";
import { WhereSubject, type WhereSubjectParams } from "./WhereSubject.js";
import type { Shipment } from "./Shipment.js";

export interface StepParams extends WhereSubjectParams {
  arrivalDate?: string;
  duration?: string;
  isStepOf?: string;
  delivery?: Shipment;
  pickUp?: Shipment;
}
export class Step extends WhereSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Step";
  }

  arrivalDate?: string;
  duration?: string;
  isStepOf?: string;
  delivery?: Shipment;
  pickUp?: Shipment;

  constructor(
    semanticId: string,
    params?: StepParams,
  ) {
    super(semanticId, params);
    this.arrivalDate = params?.arrivalDate;
    this.duration = params?.duration;
    this.isStepOf = params?.isStepOf;
    this.delivery = params?.delivery;
    this.pickUp = params?.pickUp;
    this.semanticType = Step.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:arrivalDate", () => this.arrivalDate);
    this.registerSemanticProperty("dfc-b:duration", () => this.duration);
    this.registerSemanticProperty("dfc-b:isStepOf", () => this.isStepOf);
    this.registerSemanticProperty("dfc-b:delivery", () => this.delivery);
    this.registerSemanticProperty("dfc-b:pickUp", () => this.pickUp);
  }
  static {
    SemanticObject.typeRegistry.set(Step.SEMANTIC_TYPE, Step);
  }
}

File: ruby-gem/lib/models/step.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Step
require_relative '../core/semantic_object'
require_relative 'where_subject'


module DfcLinkmlConnector
  module Models
    class Step < WhereSubject
      SEMANTIC_TYPE = "dfc-b:Step".freeze

      # @return [String]
      attr_accessor :arrival_date

      # @return [String]
      attr_accessor :duration

      # @return [String]
      attr_accessor :is_step_of

      # @return [Shipment]
      attr_accessor :delivery

      # @return [Shipment]
      attr_accessor :pick_up

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, arrivalDate: nil, duration: nil, isStepOf: nil, delivery: nil, pickUp: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, arrivalDate: nil, duration: nil, isStepOf: nil, delivery: nil, pickUp: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @arrival_date = arrivalDate
        @duration = duration
        @is_step_of = isStepOf
        @delivery = delivery
        @pick_up = pickUp
        self.semanticType = "dfc-b:Step"
        registerSemanticProperty("dfc-b:Step:arrival_date", &method("arrival_date")).valueSetter = method("arrival_date=")
        registerSemanticProperty("dfc-b:Step:duration", &method("duration")).valueSetter = method("duration=")
        registerSemanticProperty("dfc-b:Step:is_step_of", &method("is_step_of")).valueSetter = method("is_step_of=")
        registerSemanticProperty("dfc-b:Step:delivery", &method("delivery")).valueSetter = method("delivery=")
        registerSemanticProperty("dfc-b:Step:pick_up", &method("pick_up")).valueSetter = method("pick_up=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
